espnet2/asr/frontend/sinc.py

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/env python3
+#  2020, Technische Universität München;  Ludwig Kürzinger
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+"""Sinc convolutions for raw audio input."""
+
+from collections import OrderedDict
+from espnet2.asr.frontend.abs_frontend import AbsFrontend
+from espnet2.layers.sinc_conv import LogCompression
+from espnet2.layers.sinc_conv import SincConv
+import humanfriendly
+import torch
+from typing import Optional
+from typing import Tuple
+from typing import Union
+
+
+class LightweightSincConvs(AbsFrontend):
+    """Lightweight Sinc Convolutions.
+
+    Provide a frontend for raw audio input.
+    https://arxiv.org/abs/2010.07597
+    """
+
+    def __init__(
+        self,
+        fs: Union[int, str, float] = 16000,
+        in_channels: int = 1,
+        out_channels: int = 256,
+        activation_type: str = "leakyrelu",
+        dropout_type: str = "dropout",
+    ):
+        """Initialize the module.
+
+        Args:
+            fs: Sample rate.
+            in_channels: Number of input channels.
+            out_channels: Number of output channels (for each input channel).
+            activation_type: Choice of activation function.
+            dropout_type: Choice of dropout function.
+        """
```

Comment:
please add  `assert check_argument_types()` for all __init__()

Reply:
Added